internal/service/handlers/verify_passport.go

```diff
@@ -235,27 +284,48 @@ func claimReferralSpecificEvents(r *http.Request, balance data.Balance, autoClai
 		return fmt.Errorf("get fulfilled referral specific events: %w", err)
 	}
 
-	if len(events) == 0 {
+	// Specify how many events can be claimed
+	countToClaim := int64(len(events))
+	if countToClaim == 0 {
 		return nil
 	}
 
-	if !country.ReserveAllowed || country.Reserved >= country.ReserveLimit {
-		return nil
+	// If, for example, 10 points are awarded for an event,
+	// and 2 points remain before reaching the reservation
+	// limit, then this event can be claimed. And since there
+	// can be many events with invited friends, need to calculate
+	// the maximum number of events that can be claimed in order
+	// not to exceed the limit.
+	if country.Reserved+countToClaim*evTypeRef.Reward >= country.ReserveLimit+evTypeRef.Reward {
+		countToClaim = int64(math.Ceil(float64(country.ReserveLimit-country.Reserved) / float64(evTypeRef.Reward)))
 	}
 
-	for _, event := range events {
-		if _, err = claimEventWithPoints(r, event, evTypeRef.Reward, &balance); err != nil {
-			return fmt.Errorf("failed to claim referral specific event: %w", err)
-		}
+	eventsToClaimed := make([]string, countToClaim)
+	for i := 0; i < int(countToClaim); i++ {
+		eventsToClaimed[i] = events[i].ID
```

Comment:
Index always be in range because of check: 
`	if country.Reserved+countToClaim*evTypeRef.Reward >= country.ReserveLimit+evTypeRef.Reward {
		countToClaim = int64(math.Ceil(float64(country.ReserveLimit-country.Reserved) / float64(evTypeRef.Reward)))
	}`